Instruction for installation on OS X

xenatisch opened the pull request:
The process tested on OS X 10.11 was thoroughly correct. However, iRuby would not appear in the list of available kernels if the notebook is opened using:
`
jupyter notebook
`
The work around is to start the notebook for the first time using:
`
iruby notebook
`
This way, it will be added to the list of kernels permanently, so that subsequently, the notebook can be opened using the former command.

minad committed:
check number of arguments to the kernel

minad committed:
use backticks to raise exceptions

minad committed:
log kernel exceptions

minad committed:
fix command line arguments to allow "iruby --log-level=DEBUG"

minad committed:
fix #12

minad committed:
v0.1.8

minad committed:
installation instructions

minad committed:
handle shutdown request

minad committed:
contributors

minad committed:
add ipython version check

minad committed:
v0.1.9

agarie committed:
Allow LaTeX display for NMatrix

This commit adds support for printing NMatrices via LaTeX. Not very much
to explain!

jvns committed:
Add note about notebook needing libzmq 3.2

minad committed:
Merge pull request #16 from jvns/patch-2

Add note about notebook needing libzmq 3.2

minad committed:
add note about ipython version

minad committed:
fix #19 pp, missing <<

translunar committed:
Fixed spelling on one error message and clarified an exception about
pry. Fixes #20.

minad committed:
Merge pull request #21 from mohawkjohn/errors

Fixed spelling on one error message and clarified an exception about pry.

minad committed:
Merge pull request #24 from agarie/nmatrix_to_latex

Allow LaTeX display for NMatrix

minad committed:
fix symlink issue

minad committed:
v0.1.10

minad committed:
push pry binding if stack is empty

minad committed:
v0.1.11

minad committed:
IRuby#table add option maxrows

IRuby.table(..., maxrows: 10)

minad committed:
flexible display implementation

supports multiple formats per datatype

closes #25

minad committed:
Add IRuby#javascript

IRuby.javascript('alert("Hello, world");')

minad committed:
Add IRuby#svg

IRuby.svg(%{<svg
xmlns="http://www.w3.org/2000/svg"xmlns:xlink="http://www.w3.org/1999/xlink"><rect
x="10" y="10" height="100" width="100"style="stroke:#ff0000; fill:

minad committed:
powerful display system with format and datatype registry

minad committed:
changes

minad committed:
v0.1.12

minad committed:
replace IPython in title with IRuby

minad committed:
Merge branch 'master' of github.com:minad/iruby

minad committed:
latex output for Complex

minad committed:
improved IRuby.table, supports :maxrows, :maxcols

minad committed:
table output

minad committed:
IRuby.table: Better handling of Hash of Array

minad committed:
table examples

minad committed:
workaround application/javascript output is not stored in the notebook
or not executed

https://github.com/ipython/ipython/issues/6259

minad committed:
changes

minad committed:
v0.1.13

domitry committed:
windows compatibility

zalt50 committed:
Fix typo

minad committed:
Merge pull request #35 from zalt50/prototype

Fix typo Magick

zalt50 committed:
NArray#transpose requires arguments.

minad committed:
Merge pull request #36 from zalt50/fix_displaying_narray

NArray#transpose requires arguments.

jnv committed:
Allow empty argument for OStream#puts

jnv committed:
Add OStream#sync accessor for IO compatibility

minad committed:
Merge pull request #40 from jnv/ostream

Improve OStream compatibility with IO

domitry committed:
Added Comm support

domitry committed:
Add IRuby::Widget

domitry committed:
Enable Widget to handle comm message from backbone

domitry committed:
Merge pull request #43 from domitry/widget_support

Widget support

zorbash committed:
Replace Matrix#row_count with Matrix#row_size

minad committed:
Merge pull request #44 from Zorbash/master

Replace Matrix#row_count with Matrix#row_size

minad committed:
update contributors

minad committed:
update readme

minad committed:
fix links

minad committed:
rename for consistency

minad committed:
move license

minad committed:
update readme

minad committed:
link

minad committed:
update links

spacewander committed:
conflate conditions with same actions

The condition with `/comm/` is added in commit b2a6002.
After reading the commit, I think it can be conflated with the one
above.

minad committed:
Merge pull request #8 from spacewander/conflation

conflate conditions with same actions

liudangyi committed:
refactor command.rb to support ipython3

1. Add register / unregister method to command line.
2. Now IRuby and IPython use the same profile directory.
3. Remove a warning from kernel.rb

minad committed:
add note to readme

liudangyi committed:
Better logs

liudangyi committed:
Support logging to multiple files

liudangyi committed:
Use MultiLogger instead of Logger. Big mistake.

liudangyi committed:
Add test for multilogger

domitry committed:
Fix #15

The first and last lines are just for backward compativility.
If those lines are deleted, a white output-area do not appear when Ostream#puts is called with no arguments.

minad committed:
Merge pull request #14 from liudangyi/better_logs

Better logs

minad committed:
stylistic changes

minad committed:
add todo

minad committed:
add todo

minad committed:
pip/virtualenv installation instruction

minad committed:
fix

minad committed:
logger allows constructor arguments

minad committed:
update deps

minad committed:
switch to rbczmq

minad committed:
switch to new ipython3 protocol

minad committed:
fix

minad committed:
static isn't used as of now

minad committed:
we will move widgets to a separate gem

minad committed:
refactor comm system slightly

minad committed:
send ident

minad committed:
refactor session

minad committed:
wrap sockets in session

minad committed:
fixes

minad committed:
fixes

minad committed:
terminate kernel if SystemExit is caught

minad committed:
Revert "static isn't used as of now"

This reverts commit 55274d75650a5d3eb0820625eab2c20b5718e953.

minad committed:
move static out of lib

minad committed:
prepare 0.2.0

minad committed:
update example notebook

minad committed:
protocol update

minad committed:
fix completion cursor position

minad committed:
get the tests running

minad committed:
attic

minad committed:
minor display fixes

minad committed:
update readme

minad committed:
resolved https://github.com/ipython/ipython/issues/6259

minad committed:
require ruby 2.0

minad committed:
don't hide banner on cmdline

minad committed:
update contributors

minad committed:
enable travis.yml

minad committed:
travis

minad committed:
longer first except timeout

minad committed:
travis: use virtualenv

minad committed:
Revert "travis: use virtualenv"

This reverts commit bca80c9cc9e226ec8c5cfd67ae1f1c14f42aaefd.

minad committed:
travis install libzmq

minad committed:
travis libzmq3 is already there

minad committed:
faster iruby boot

minad committed:
update readme

minad committed:
logo directory

minad committed:
v0.2.0 (IPython3/Juypter compatible)

minad committed:
changes v0.2.0

minad committed:
changes

minad committed:
update readme

minad committed:
travis.yml - add notifications

minad committed:
remove pry-theme

minad committed:
add iruby logo

minad committed:
more verbose help

minad committed:
v0.2.1

minad committed:
more version information

minad committed:
add ruby version to kernel.json

minad committed:
wrap all zmq stuff in IRuby::Session

minad committed:
fix connect_reply

minad committed:
implement iruby history variables In, Out etc

minad committed:
fix history

minad committed:
fix history

minad committed:
history: populate _i and _ii later

minad committed:
use store_history flag

minad committed:
fixed inspect_request

minad committed:
code style

minad committed:
refactor how idents are handled

minad committed:
fixes

minad committed:
history example

minad committed:
update examples

minad committed:
changes

minad committed:
HTML table formatter - cells are converted to html

minad committed:
slight refactor of comm system

minad committed:
v0.2.2

minad committed:
placeholder for kernel css and kernel.js

minad committed:
mention try.jupyter.org

minad committed:
require bond only if PlainBackend is used

minad committed:
set code mirror indentUnit = 2 in kernel.js, fix #27

minad committed:
fix #26, only complete last line

minad committed:
v0.2.3

minad committed:
kernel.js use strict

minad committed:
command: use FileUtils.copy list

minad committed:
fix timeout issue #19

minad committed:
fix loggers

minad committed:
code formatting

minad committed:
use correct mime type for ruby

minad committed:
update notebooks

minad committed:
better exception handling, fix #17

minad committed:
v0.2.4

minad committed:
fix #29

minad committed:
pry - reset last result before evaluation

minad committed:
correct ruby version. we require ruby >= 2.1

minad committed:
remove notebooks, there are now at sciruby/sciruby-notebooks, remove
obsolete stuff

minad committed:
update link

minad committed:
image link

minad committed:
image link

minad committed:
convert readme to markdown, sorry rdoc

minad committed:
image

minad committed:
remove old readme

minad committed:
update readme

minad committed:
update readme

minad committed:
update license

minad committed:
changes

minad committed:
add timestamps to changelog

    for i j in $(git log --oneline --pretty='format:%d %ci' | grep
'(tag: ' | sed -r 's/.*\(tag: v([^,]*).*\) ([0-9-]*) .*/\1 \2/g'); do
            sed -e "s/^$i$/$i ($j)/g" CHANGES > CHANGES.2
            mv CHANGES.2 CHANGES
    done

minad committed:
changes

minad committed:
update readme

minad committed:
move iruby utils to Utils module, add tex alias for latex

minad committed:
changes

hainesr committed:
Don't override terminal options for gnuplot display.

Allow the use of more options to the terminal command while ensuring that the
terminal type is always set to svg and that enhanced mode is chosen unless
explicitly turned off. This preserves the expected defaults provided by the
previous implementation.

minad committed:
Merge pull request #33 from hainesr/gnuplot-terminal

Don't override terminal options for Gnuplot display.

minad committed:
code style

minad committed:
v0.2.5

minad committed:
clean gemspec

minad committed:
support to_tex

minad committed:
standard names

minad committed:
use require_relative

minad committed:
check registered kernel and gemfile

should fix #36 and #31

minad committed:
v0.2.6

minad committed:
catch all exceptions in Display type method

minad committed:
v0.2.7

minad committed:
fix #38

domitry committed:
Windows compatibility

Almost same as I did in v0.2.7
(fda30d610786db4e2929554de37ba25972846f36)

domitry committed:
Add session_ffi.rb

Just copied 3 methods from lib/iruby/session.rb.

domitry committed:
Use constants defined in the module ZMQ

domitry committed:
Use ZMQ::Device instead of ZMQ.proxy

ZMQ.proxy is not implemented in ffi-rzmq

domitry committed:
Use ZMQ::Socket#send_string instead of #send_message

ZMQ::Socket#send_message is not implimented in ffi-rzmq.

domitry committed:
Swap arguments

domitry committed:
Use ZMQ::Socket#recv_string instead of #recv_message

ZMQ::Socket#recv_message is not implemented in ffi-rzmq.

domitry committed:
Session#serialize now return Array instead of ZMQ::Message

domitry committed:
Use ffi-rzmq if require 'rbczmq' fails

domitry committed:
Remove rbczmq from dependency

I don't know what is the best solution about add_runtime_dependency.
If this line is not removed Windows users will very confused, but if not
users will wonder how to solve LoadError 'ffi-rzmq' or 'rbczmq'.

domitry committed:
Merge pull request #49 from domitry/windows

Windows compatibility

domitry committed:
Separate IRuby::Session into two files.

One is for rbczmq and the other is for ffi-rzmq.

domitry committed:
Do not catch LoadError as e

domitry committed:
Move 2 methods in session/rbczmq common with ffi_rzmq to session/base

domitry committed:
Use inheritance instead of mix-in

domitry committed:
Unify the style of require_relative

domitry committed:
Move 'Quick start' section of now to 'Quick start => Ubuntu/Debian'.

domitry committed:
Add the installation method for Windows

domitry committed:
Add the installation method for Mac

domitry committed:
Add the section 'After the installation'

domitry committed:
Remove libzmq from the section 'dependency' and add an annotation

domitry committed:
Fix typo

domitry committed:
Move Session#serialize to SessionBase

domitry committed:
Use mix-in instead of inheritance

minad committed:
Merge pull request #51 from domitry/readme

Readme

minad committed:
Merge pull request #50 from domitry/compatibility_with_ffi_rzmq

Compatibility with ffi rzmq

minad committed:
rename base to SessionSerialize

minad committed:
remove unnecessary private

murashin committed:
add installation document for FreeBSD

minad committed:
Merge pull request #53 from murashin/doc_freebsd

add installation document for FreeBSD

domitry committed:
Fix indent of the code embeded in README

domitry committed:
v0.2.8

domitry committed:
Add description about v0.2.8

Commit:
Add clear_output display function

domitry committed:
Merge pull request #55 from mrkn/clear_output

Add clear_output display function

minrk commented:
I think you made a PR against the wrong repo. IRuby now lives here: https://github.com/sciruby/iruby

generall committed:
Update README.md

jRuby installation instruction.

domitry committed:
Merge pull request #57 from generall/patch-1

Update README.md

kozo2 committed:
add a note about DevKit for Windows

domitry committed:
Merge pull request #58 from kozo2/patch-1

add a note about DevKit for Windows

kozo2 committed:
describe explicit version and exe address

domitry committed:
Merge pull request #59 from kozo2/patch-2

describe explicit version and exe address

kylekyle committed:
patches PryBackend to throw unterminated string and unexpected end-of-file syntax errors

minad committed:
Merge pull request #63 from kylekyle/master

PryBackend syntax errors

domitry committed:
Update Quick Start for Mac OS X users

domitry committed:
Update README for Mac users

ref: http://koki-h.hatenablog.com/entry/2016/02/21/193307

CGamesPlay committed:
Convert Set to Array before using as header

This is needed in cases where a table rendered using an Array of Hashes has more than `maxcols` columns. The error is `NoMethodError` with `Set#[]` on line 83, where header is indexed into.

minad committed:
Merge pull request #75 from CGamesPlay/patch-1

Convert Set to Array before using as header

kylekyle committed:
IRuby::Input (#77)

Adds the IRuby::Input class which provides widgets for getting inputs from the user.

v0dro committed:
update changelog

v0dro committed:
update version

kylekyle committed:
Fixes #83

kylekyle committed:
Use less memory

Pry keeps the commands and outputs for the last 100 evals in memory. This makes notebooks with large outputs (like images) crawl to a halt. Since IRuby keeps a history of 3, pry should as well. I would love to set memory_size to 0, but I'm not sure how that would effect the IRuby::History module.

kylekyle committed:
Fix z-index

The datepicker sets a z-index of 3, which is way too low in bootstrap. The !important tag makes sure this bigger z-index is used across all browsers.

kylekyle committed:
fixes #85 and fixes #86

kylekyle committed:
Process correct key

kylekyle committed:
fixes #89

kylekyle committed:
added defaults, fixes #88

kylekyle committed:
added multiple widget, fixes #87

kylekyle committed:
added multiple widget

kylekyle committed:
fixes #94
